networkit/cpp/embedding/AliasSampler.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+ * AliasSampler.cpp
+ *
+ *  Created on: 03.07.2020
+ *      Author: Klaus Ahrens  <ahrens@informatik.hu-berlin.de>
+ *
+ *  adapted and reimplemented in C++17
+ *  from node2vec [https://arxiv.org/pdf/1607.00653v1.pdf]
+ *
+ */
+
+// networkit-format
+
+#include <mutex>
+#include <random>
+#include <vector>
+
+#include <networkit/Globals.hpp>
+#include <networkit/auxiliary/Random.hpp>
+
+#include "AliasSampler.hpp"
+
+namespace NetworKit {
+namespace Embedding {
+
+std::mt19937_64 &getURNG() {
+    static bool seedPerThread = true;
+    static std::once_flag once;
+    std::call_once(once, [] { Aux::Random::setSeed(1, seedPerThread); });
+
+    return Aux::Random::getURNG();
```

Comment:
Why do you set a custom seed here? And why the static seed? This makes reproducible experiments or running experiments with different seeds completely impossible as it is impossible to run the algorithm twice with the same seed or an explicitly set seed.